lib/db/schema/patch-034-035.sql

```diff
@@ -0,0 +1,87 @@
+CREATE TABLE IF NOT EXISTS unblockCodes (
+  uid BINARY(16) NOT NULL,
+  unblockCodeHash BINARY(32) NOT NULL,
+  createdAt BIGINT SIGNED NOT NULL,
+  PRIMARY KEY(uid, unblockCodeHash)
+) ENGINE=InnoDB;
+
+CREATE PROCEDURE `createUnblockCode_1` (
+    IN inUid BINARY(16),
+    IN inCode BINARY(32),
+    IN inCreatedAt BIGINT SIGNED
+)
+BEGIN
+    INSERT INTO unblockCodes(
+        uid,
+        unblockCodeHash,
+        createdAt
+    )
+    VALUES(
+        inUid,
+        inCode,
+        inCreatedAt
+    );
+END;
+
+CREATE PROCEDURE `consumeUnblockCode_1` (
+    inUid BINARY(16),
+    inCode BINARY(32)
+)
+BEGIN
+    CREATE TEMPORARY TABLE IF NOT EXISTS unblockTimestamps AS
+    (
+        SELECT createdAt FROM unblockCodes
+        WHERE
+            uid = inUid
+        AND
+            unblockCodeHash = inCode
+    );
```

Comment:
I remain unenthused about the use of a temporary table here, but will defer to your judgement.  Some things to sanity-check:

* The user must have permission to created temporary tables, we'll need to confirm with @jrgm if this is OK in production
* According to [1] a temporary table lives until the owning session is closed.  We use connection pooling which means we keep the same mysql session alive across different web requests.  So you may need to explicitly drop this table when you're done with it, otherwise all subsequent calls to `consumeUnblockCode` will return whatever value of `createdAt` was left in the table, even if the code didn't match anything.

[1] https://dev.mysql.com/doc/refman/5.7/en/create-table.html